pallets/users/src/lib.rs

```diff
@@ -32,60 +35,83 @@ pub mod pallet {
 
 	#[pallet::storage]
 	#[pallet::getter(fn something)]
-
+	// define pallet storage abilities up top and create function definitions inside
 	pub type Something<T> = StorageValue<_, u32>;
 
-
 	#[pallet::event]
 	#[pallet::metadata(T::AccountId = "AccountId")]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
-
 		SomethingStored(u32, T::AccountId),
+		UserCreated(T::AccountId),
 	}
 
+	#[pallet::storage]
+	// users store
+	pub type Users<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, User>;
+
+	use v1::User;
+	pub mod v1 {
+		use codec::{Decode, Encode};
+		use sp_std::vec::Vec;
+
+		#[derive(Encode, Decode, Default, Clone)]
+		pub struct User {
+			pub rank_points: u32,
+		}
+	}
 
 	#[pallet::error]
 	pub enum Error<T> {
 		/// No Value
 		NoneValue,
 		/// Storage is overflow
 		StorageOverflow,
+		/// User already exists
+		UserAlreadyExists,
 	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
-
 		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
 		pub fn do_something(origin: OriginFor<T>, something: u32) -> DispatchResult {
-
 			let who = ensure_signed(origin)?;
 
-
 			<Something<T>>::put(something);
 
-
 			Self::deposit_event(Event::SomethingStored(something, who));
 
 			Ok(())
 		}
 
+		// use base weight then add on any additional transactions
+		// This pallet simply returns back the message in the event
+		// strings are u8 arrays. utf-8
+		/// Signed transaction to create user
+		#[pallet::weight(0 + T::DbWeight::get().writes(1))]
+		pub fn make_user(origin: OriginFor<T>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			ensure!(!Users::<T>::contains_key(&who), Error::<T>::UserAlreadyExists);
+			<Users<T>>::insert(&who, User { rank_points: 0 });
+
+			Self::deposit_event(Event::UserCreated( who));
+
+			Ok(())
+		}
 
 		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
 		pub fn cause_error(origin: OriginFor<T>) -> DispatchResult {
```

Comment:
I don't think we need this, right?

Reply:
Yeah, I'll change it when there's power. I think a more efficient way would be to create records on the fly in the same dispatch